spark/src/test/scala/org/apache/comet/CometExpressionSuite.scala

```diff
@@ -2390,4 +2390,14 @@ class CometExpressionSuite extends CometTestBase with AdaptiveSparkPlanHelper {
       checkSparkAnswer(df.select("arrUnsupportedArgs"))
     }
   }
+
+  test("array_contains") {
+    withTempDir { dir =>
+      val path = new Path(dir.toURI.toString, "test.parquet")
+      makeParquetFileAllTypes(path, dictionaryEnabled = false, n = 10000)
+      spark.read.parquet(path.toString).createOrReplaceTempView("t1");
+      checkSparkAnswerAndOperator(
+        spark.sql("SELECT array_contains(array(_2, _3, _4), _2) FROM t1"))
```

Comment:
Could you also add tests for the following cases:

- array is null
- key is null

You can use a `CASE` expression to insert null values for array and key. Here is an example from another test:

```scala
        checkSparkAnswerAndOperator(
          spark.sql("SELECT array_append((CASE WHEN _2 =_3 THEN array(_4) END), _4) FROM t1"));
```

Reply:
Either if array or keys is null, it fails during sparks Check Analysis. Which is why i didn't add them.